operator/v1alpha1/types.go

```diff
@@ -150,3 +150,31 @@ type DelegatedAuthorization struct {
 	// disabled indicates that authorization should be disabled.  By default it will use delegated authorization.
 	Disabled bool `json:"disabled,omitempty"`
 }
+
+// StaticPodOperatorStatus is status for controllers that manage static pods.  There are different needs because individual
+// node status must be tracked.
+type StaticPodOperatorStatus struct {
+	OperatorStatus `json:",inline"`
+
+	// latestDeploymentGeneration is the deploymentID of the most recent deployment
+	LatestDeploymentGeneration int32 `json:"latestDeploymentGeneration"`
+
+	// nodeStatuses track the deployment values and errors across individual nodes
+	NodeStatuses []NodeStatus `json:"nodeStatuses"`
+}
+
+// NodeStatus provides information about the current state of a particular node managed by this operator.
+type NodeStatus struct {
+	// nodeName is the name of the node
+	NodeName string `json:"nodeName"`
+
+	// currentDeploymentGeneration is the ID of the most recently successful deployment
+	CurrentDeploymentGeneration int32 `json:"currentDeploymentGeneration"`
+	// targetDeploymentGeneration is the ID of the deployment we're trying to apply
+	TargetDeploymentGeneration int32 `json:"targetDeploymentGeneration"`
+	// lastFailedDeploymentGeneration is the ID of the deployment we tried and failed to deploy.
+	LastFailedDeploymentGeneration int32 `json:"lastFailedDeploymentGeneration"`
```

Comment:
What do you expect from this value? Why do we need it? 

I would have expected current + target. If `len(Errors) > 0`, it's about the target.

Reply:
> What do you expect from this value? Why do we need it?
> 
> I would have expected current + target. If `len(Errors) > 0`, it's about the target.

If the latestDeployment moves again, you should try to install it. However, you don't want to stop an install part way.  So you need to be able to determine.
 1. steady state
 2. transitioning state
 3. transition is finished and not successful.